src/sentry/incidents/logic.py

```diff
@@ -1125,6 +1125,11 @@ def delete_alert_rule_trigger(trigger: AlertRuleTrigger) -> None:
     """
     Deletes an AlertRuleTrigger
     """
+    from sentry.workflow_engine.migration_helpers.alert_rule import (
+        dual_delete_migrated_alert_rule_trigger,
+    )
+
+    dual_delete_migrated_alert_rule_trigger(trigger)
```

Comment:
I initially did this too in https://github.com/getsentry/sentry/pull/83562#discussion_r1917514332 but Josh had me move it to where it's being called in the serializer. In this case you can put it in https://github.com/getsentry/sentry/blob/master/src/sentry/incidents/endpoints/organization_alert_rule_details.py#L106-L111